### ISSUE TITLE
identifier encoding causes parse error

### ISSUE THREAD

**Issue by jdalton** [meriyah/meriyah#221]
This is an entry from a test262 test:

```js
headLhsAsyncEscaped: {
  let async;
  for (\u0061sync of [7]);
  assert.sameValue(async, 7);
}
```
Meriyah incorrectly throws:
Unexpected token: 'escaped keyword'

**Comment by 3cp** [comment]
Released in v4.3.1.

**Comment by jdalton** [comment]
Thank you @3cp!
